app/services/exchange_variant_deleter.rb

```diff
@@ -0,0 +1,8 @@
+class ExchangeVariantDeleter
+  def delete(product)
+    variant_ids = product.variants.map(&:id)
+    ExchangeVariant.
+      where(variant_id: variant_ids).
```

Comment:
You can avoid a query here:
```suggestion
      where(variant_id: product.variants.select(:id)).
```